src/BoilerPlate/Asteroids.cpp

```diff
@@ -0,0 +1,84 @@
+#include "Asteroids.hpp"
+const float SPACE_MAGIC = 30.0f;
+const float MAX_SPEED = 60.0f;
+const float ASTEROID_RADIO = 27.0f;
+const float ASTEROID_ADVANCE_METERS = 10.0f;
+const float STARTING_ASTEROID_ANGLE = 30.0f;
+const float ZERO = 0.0f;
+const float ASTEROID_ROTATE_SPEED = 0.3f;
+float ASTEROID_ROTATE_ANGLE = 0.0f;
+
+
+Asteroids::Asteroids() {}
+Asteroids::Asteroids(float width,float height):
+	Entity(width,height) {
+
+	asteroid_Size = 3;
+	angle = STARTING_ASTEROID_ANGLE;
+	PushDrawEntity();
+	mass = (float)(asteroid_Size);
+	position = Vector2((float)(rand()%1000), (float)(rand()%1000));
+	velocity = Vector2(ZERO, ZERO);
+	radius = asteroid_Size * ASTEROID_RADIO;
+}
+Asteroids::Asteroids(Asteroid_Size size, float width, float height):
+	Entity(width, height) {
+	angle = STARTING_ASTEROID_ANGLE;
+	asteroid_Size = size; 
+	mass = (float)(asteroid_Size);
+	PushDrawEntity();
+	position = Vector2((float)(rand() % 1000), (float)(rand() % 1000));
+	velocity = Vector2(ZERO, ZERO);
+	radius = asteroid_Size * ASTEROID_RADIO;
+}
+Asteroids::Asteroids(Asteroid_Size size, Asteroids rock) :
+	Entity(rock.maxWidth * 2, rock.maxHeight * 2) {
+	angle = rock.angle + rand() % 360;
+	asteroid_Size = size;
+	mass = (float)(asteroid_Size);
+	PushDrawEntity();
+	position = Vector2(rock.position.x,rock.position.y);
+	velocity = Vector2(ZERO, ZERO);
+	radius = (float)(asteroid_Size) * ASTEROID_RADIO;
+}
+int Asteroids::Asteroid_GetSize() {
+	return asteroid_Size;
+}
+void Asteroids::Update( float DT ) {
+	ASTEROID_ROTATE_ANGLE += ASTEROID_ROTATE_SPEED;
+	ApplyImpulse(Vector2(SPACE_MAGIC, SPACE_MAGIC));
+	float speed = sqrtf(velocity.x * velocity.x + velocity.y * velocity.y);
+	MathUtilities math;
+	if (speed > MAX_SPEED) {
+		velocity.x = math.Clamp(velocity.x, speed, MAX_SPEED);
+		velocity.y = math.Clamp(velocity.y, speed, MAX_SPEED);
+		speed = MAX_SPEED;
+	}
+	Entity::Update(DT);
+}
+void Asteroids::PushDrawEntity() {
+	int size = Asteroid_GetSize();
+
+	pointsContainer.push_back(Vector2(-26.0f * size, 6.0f  * size));//a
```

Comment:
Still unsure on why you're using these magic numbers here. No documentation about them, don't know why they exist.